lib/util.js

```diff
@@ -15,11 +15,25 @@ function getPaths(publicPath, compiler, url) {
   const compilers = compiler && compiler.compilers;
   if (Array.isArray(compilers)) {
     let compilerPublicPath;
+
+    // the path portion of compilerPublicPath
+    let compilerPublicPathBase;
+
     for (let i = 0; i < compilers.length; i++) {
       compilerPublicPath = compilers[i].options
     && compilers[i].options.output
     && compilers[i].options.output.publicPath;
-      if (url.indexOf(compilerPublicPath) === 0) {
+
+      // handle the case where compilerPublicPath is a URL with hostname
+      if (compilerPublicPath.startsWith('/')) {
+        compilerPublicPathBase = compilerPublicPath;
+      } else {
+        const urlObject = parse(compilerPublicPath);
```

Comment:
this could be stripped (optional) 

Used once, so: `compilerPublicPathBase = parse(compilerPublicPath).pathname;` and rather add a comment above?